scripts/utils/parsing.py

```diff
@@ -0,0 +1,96 @@
+"""Parsing-related utility functions."""
+
+from __future__ import annotations
+
+import ast
+from pathlib import Path
+from typing import List
+
+
+def _get_ast_tree(file_path: Path) -> ast.AST:
+    """Get the parsed AST tree for a Python file.
+    
+    Args:
+        file_path: Path to the Python file to parse.
+    
+    Returns:
+        The parsed AST tree.
+    """
+    with open(file_path, 'r', encoding='utf-8') as f:
+        source = f.read()
+    return ast.parse(source)
+
+
+def _is_target_decorator(decorator: ast.AST, decorator_type: str) -> bool:
```

Comment:
same comment as _get_ast_tree()
```suggestion
def is_target_decorator(decorator: ast.AST, decorator_type: str) -> bool:
```